ams/server/src/main/java/com/netease/arctic/server/manager/SpiPluginManager.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.manager;
+
+import com.netease.arctic.ams.api.AmoroPlugin;
+import com.netease.arctic.ams.api.PluginManager;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Plugin manager based on SPI
+ * @param <T> plugin type
+ */
+public abstract class SpiPluginManager<T extends AmoroPlugin> implements PluginManager<T> {
```

Comment:
I think there are two important features of the plugin mechanism: extensibility and class isolation. Is there a lack of loader isolation related design?

Reply:
this could be a future work, Class isolation is not necessary for current manangers